doc/pins/p4rt_app_hld.md

```diff
@@ -0,0 +1,169 @@
+   acl_ingress_meter.read(local_metadata.color);
+   acl_ingress_counter.count();
+ }
+```
+
+similarly a match qualifier for IPv4 packets may use a @sai_field like:
+
+```
+// Model the ACL Ingress stage.
+table acl_ingress_table {
+   key = {
+     // Allow flows to match on IPv4 packets. ACL qualifiers are marked 'optional'
+     // and will be ignored if unset.
+     //
+     // If set and the packet has a valid IPv4 header when reaching the ACL Ingress
+     // stage the qualifier will match.
+     headers.ipv4.isValid() : optional @name("is_ipv4") @id(2)
+         @sai_field(SAI_ACL_TABLE_ATTR_FIELD_ACL_IP_TYPE/IPV4ANY);
+   }
+}
+```
+
+
+### Writing (insert/modify/delete) into the P4RT APPL_DB Table
+
+P4 Runtime uses a Platform Independent (PI) request format for all writes & reads. The P4RT application is responsible for translating these requests into APPL_DB entries according to the AppDb schema (covered later). Table entry updates are only allowed from a primary connection. These table entries can further be restricted based on controller roles. Reads can be done from either a primary or backup connection. The entries can also be restricted based on roles.
+
+
+### WCMP/Hashing
+
+WCMP group enables the weight adjusted selection of members as an action for a route entry match. Depending on the hash value computed from selected fields of the packet header, one of the members is chosen for packet forwarding.
+
+There are 2 parts to the WCMP feature in P4RT, first is the selection of the packet header fields for different packets, the hashing algorithm to be used etc and the second is the actual WCMP group, member and route entry definitions. The values for the former come in the P4Info file and the latter in the Write request.
+
+OrchAgent added a new orchestrator HashOrch to take care of programming the hash related SAI attributes in the SWITCH_TABLE. The APP_DB format for hashing entry is described in the P4RT Schema HLD.
+
+
+### Response path
+
+P4RT application needs a response status for every request (success and failure) that is sent southwards to the OrchAgent. This ensures the external client making the request is aware of the status and can respond appropriately.
+
+At a high level, syncd operates in synchronous mode and OrchAgent relays the hardware status of the operation back to P4RT application via a separate notification channel and writes all successful responses into the APPL_STATE_DB. This abstraction differs from the existing STATE_DB which does not provide an application level response. P4RT application uses the APPL_STATE_DB to restore the entry in APPL_DB when a particular request fails. More details on the response path design is captured in the AppDb Shema HLD.
+
+
+## APPL DB Schema High-Level Design
+
+P4RT application introduces new tables that are written to APPL_DB for the tables it intends to program via the P4Orch. The schema of the P4RT tables and other details are captured in the P4RT DB Schema HLD.
```

Comment:
Could you provide the link to "P4RT DB Schema HLD"? Reminder that we also need Yang model defined for new schema.